validator.go

```diff
@@ -0,0 +1,201 @@
+package jwt
+
+import (
+	"crypto/subtle"
+	"fmt"
+	"time"
+)
+
+// Validator is the core of the new Validation API. It is
+type Validator struct {
+	// leeway is an optional leeway that can be provided to account for clock skew.
+	leeway time.Duration
+
+	// timeFunc is used to supply the current time that is needed for
+	// validation. If unspecified, this defaults to time.Now.
+	timeFunc func() time.Time
+
+	// verifyIat specifies whether the iat (Issued At) claim will be verified.
+	// According to https://www.rfc-editor.org/rfc/rfc7519#section-4.1.6 this
+	// only specifies the age of the token, but no validation check is
+	// necessary. However, if wanted, it can be checked if the iat is
+	// unrealistic, i.e., in the future.
+	verifyIat bool
+
+	// expectedAud contains the audiences this token expects. Supplying an empty
+	// string will disable aud checking.
+	expectedAud string
```

Comment:
Perhaps this should be `[]string` or the type currently named `ClaimStrings`?

Reply:
The way I understand it is that a system that checks a JWT for its validity must identify itself with one specific audience (e.g. its server URL), so that is why it is only a single string here. The JWT itself can be sent to multiple audiences (several servers that consume it), so that is why it is an array on the claim itself.

Does that make sense?